### ISSUE TITLE
Redefined Keys: If no remaps have been made the "Export as file" command does not write a file out

### ISSUE THREAD

**Issue by tinkerer-red** [YoYoGames/GameMaker-Bugs#8108]
### Description

I was hoping there was some unlisted key binds so I would be able to modify/patch [another issue](https://github.com/YoYoGames/GameMaker-Bugs/issues/5330) But when I went to export my key binds I realized it doesn't actually do anything.
https://gyazo.com/177a3548b70068a8a3bb0b8f6af9e603.mp4
https://github.com/user-attachments/assets/6a8103dc-97e9-476d-b813-277d44cf65f1

The path I am attempting to save to contains no special char or spaces.
`C:\Users\Red\Documents\GameMakerStudio2\_tools`

### Expected Change

Clicking save should save a file.

### Steps To Reproduce

Preferences > Redefined Keys > Export as file > attempt save somewhere

### How reliably can you recreate this issue using your steps above?

Always

### Which version of GameMaker are you reporting this issue for?

2024.8.1 (Monthly)

### Which operating system(s) are you seeing the problem on?

Windows 10

### Are you running GameMaker from inside your Steam library?

None

### Sample Package Attached?

- [x] I have attached my bug package

### Sample Project Added?

- [ ] I have included a small sample project

**Comment by tinkerer-red** [comment]
Bit late for the response but can confirm this is still happening. Not sure how you couldn't reproduce it though.

![Image](https://github.com/user-attachments/assets/9c8af2ae-488f-435f-9f63-6d731dbdc749)

the save as type isnt filled out either, where i would assume it should be given the import expects a remap file type

![Image](https://github.com/user-attachments/assets/5f7cb624-bd8e-4058-8904-42107356bcad)

[ui.log](https://github.com/user-attachments/files/19235489/ui.log)

**Comment by tinkerer-red** [comment]
I don't think this is a UX issue, think it is incredibly clear that one can redefine hotkeys in the redefine keys window. Thanks for finding out why exactly this is occurring, indeed that does line up with what I was expecting was being able to export the defaults. So this lines up.

**Comment by YYDan** [comment]
I am not sure this is the correct/best fix here - what good is now being able to export an empty file? The logical result is that we get a new bug report soon, saying "none of my key bindings exported" ;)

I would expect that the Export To File button is disabled until such time that modifications have been made. When the button is disabled there should be a tooltip explaining that the button is disabled because that person has not redefined any keys yet.

Regarding the original use-case in the Description, there should be no key bindings that magically appear in the exported file that are not already shown on the Redefine Keys dialog - if there is and so this is why we're now exporting a file when no changes have been made, this is a problem in its own right and separate bug issues should be filed for this.

**Comment by tinkerer-red** [comment]
I was actually thinking the export wouldnt be a file of changes, but instead a file containing all of the default keybinds (assuming nothing was actually redefined). Which can be useful for third party tools to reference, such as making a converter for GM keybinds to be applied to VSCode. Though i can understand if the purpose was to attempt to save space, i wouldnt see that as mandatory when it comes to keybinds, given it should be a single file per user and release build (lts, monthly, beta)

If there is another file which would be better suited for this use case could someone share that information? Then this ticket could be more directed exclusively to adding a grey out and hint tooltip for un changed redefined issue?

**Comment by SmithTom6304** [comment]
@YYDan - agreed, we should disable that button when no keys are remapped.

@tinkerer-red - it sounds like you're looking for our default hotkey files, which are included with the IDE install. Take a look in the `Hotkeys` directory within your install directory, as well as the `Plugins` directory. These are provided per-plugin, so are split up across multiple files.

**Comment by SmithTom6304** [comment]
Export remaps button will now be disabled if the user has not remapped anything

**Comment by mgeddesGM** [comment]
in IDE_ v2024.1400.0.799  Runtime v2024.1400.0.804

if a user sets copy to ctrl+v the clash warning shows. if they then use one of the suggestions (ctrl+B) the export button is disabled

but if a user changed copy to ctrl+b at the start then export is enabled
